src/errors.rs

```diff
@@ -0,0 +1,8 @@
+//! Error handling.
+
+use thiserror::Error;
+
+/// Library errors
+#[derive(Error, Debug)]
+#[error("rlimts parse error: {0}")]
+pub struct RlimitsResourceError(pub(crate) String);
```

Comment:
1. There are only a few lines to impl `std::error::Error`. It will be better if we don't add `thiserror`and `mod errors`, but inline the code into `src/lib.rs`.

2. Error naming should follow the api guidelines, e.g. `ParseResourceError`.
  <https://rust-lang.github.io/api-guidelines/naming.html#names-use-a-consistent-word-order-c-word-order>

3. The error type here should be a sealed ZST but not a string. String errors may cause unexpected allocations.

```rust
pub struct ParseResourceError{
    _priv: ()
}
```

Reply:
I'm sure about 1 and 2. I'll fix them. But, in case of 3, I think that users want to know the wrong `RLIMIT_***` at the failure. For example, if an user passes the wrong rlimit flag such as `from_str("RLIMIT_WRONG")`, the error should display the `RLIMIT_WRONG` string as an error information at the failure. So, I think that `ParseResourceError` should have the element of string type.

Reply:
In case of 3, the caller has a `&str` as source. The caller can use [anyhow::Context](https://docs.rs/anyhow/1.0.32/anyhow/trait.Context.html) to pass the source string. The caller can also format a error message.
There is no need for `from_str` to generate a error message.

Reply:
I understand.